extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlApply.java

```diff
@@ -0,0 +1,144 @@
+import io.kubernetes.client.custom.V1Patch;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.util.PatchUtils;
+import io.kubernetes.client.util.Yaml;
+import io.kubernetes.client.util.generic.GenericKubernetesApi;
+import io.kubernetes.client.util.generic.KubernetesApiResponse;
+import io.kubernetes.client.util.generic.options.PatchOptions;
+
+import java.io.IOException;
+
+public class KubectlApply<ApiType extends KubernetesObject> implements Kubectl.Executable<ApiType> {
+
+    private final ApiClient apiClient;
+    private final Class<ApiType> apiTypeClass;
+    private Mode applyMode;
+
+    private String apiGroup;
+    private String apiVersion;
+    private String resourceNamePlural;
+
+    private String yamlContent;
+    private String fieldManager;
+    private boolean forceConflict;
+
+
+    KubectlApply(Class<ApiType> apiTypeClass, ApiClient apiClient) {
+        this.apiClient = apiClient;
+        this.apiTypeClass = apiTypeClass;
+
+        this.applyMode = Mode.SERVER_SIDE_APPLY;
+        this.forceConflict = false;
+    }
+
+
+    public KubectlApply<ApiType> apiGroup(String apiGroup) {
+        this.apiGroup = apiGroup;
+        return this;
+    }
+
+    public KubectlApply<ApiType> apiVersion(String apiVersion) {
+        this.apiVersion = apiVersion;
+        return this;
+    }
+
+    public KubectlApply<ApiType> resourceNamePlural(String resourceNamePlural) {
+        this.resourceNamePlural = resourceNamePlural;
+        return this;
+    }
+
+
+    public KubectlApply<ApiType> mode(Mode applyMode) {
+        // NOTE: currently only server-side-apply supported
+        this.applyMode = applyMode;
+        return this;
+    }
+
+    public KubectlApply<ApiType> fieldManager(String fieldManager) {
+        this.fieldManager = fieldManager;
+        return this;
+    }
+
+    public KubectlApply<ApiType> forceConflict() {
+        this.forceConflict = true;
+        return this;
+    }
+
+    public KubectlApply<ApiType> load(String yamlContent) {
+        this.yamlContent = yamlContent;
+        return this;
+    }
+
+    private void verifyArguments() throws KubectlException {
+
+    }
+
+    @Override
+    public ApiType execute() throws KubectlException {
+        verifyArguments();
+        GenericKubernetesApi<ApiType, KubernetesListObject> api =
+            new GenericKubernetesApi<>(
+                apiTypeClass,
+                KubernetesListObject.class,
+                apiGroup,
+                apiVersion,
+                resourceNamePlural,
+                apiClient);
+
+        final KubernetesObject obj;
+        try {
+            obj = (KubernetesObject) Yaml.load(yamlContent);
+        } catch (IOException e) {
+            throw new KubectlException(e);
+        }
+
+        final boolean isNamespaced;
+        isNamespaced = Strings.isNullOrEmpty(obj.getMetadata().getNamespace());
```

Comment:
I don't think this will work, since the absence of a namespace == `default` as far as I know for namespaced objects. We probably need to make it an explicit boolean in the apply.

Reply:
fixed